imblearn/over_sampling/_smote/base.py

```diff
@@ -12,14 +12,14 @@
 
 import numpy as np
 from scipy import sparse
-from scipy import stats
 
 from sklearn.preprocessing import OneHotEncoder, OrdinalEncoder
 from sklearn.utils import check_random_state
 from sklearn.utils import _safe_indexing
 from sklearn.utils import check_array
 from sklearn.utils.sparsefuncs_fast import csr_mean_variance_axis0
 from sklearn.utils.sparsefuncs_fast import csc_mean_variance_axis0
+from sklearn.utils.fixes import _mode
```

Comment:
Since I would like to be compatible with `1.0.2`, I think that we will need to have our own fix trying to import from `sklearn` if available and otherwise import it.

Reply:
Should I implement an `imblearn.utils.fixes._mode` in this PR?

I traced out a `scipy.mode` implementation earlier, but it was ~600 LOC. Porting the `_mode` fix similar to `sklearn` would be faster.

Reply:
The idea is to add `imblearn.utils.fixes._mode` and copy-paste the fix from scikit-learn.
I assume that we added in 1.2. So the idea will be to import it `from sklearn.utils.fixes import _mode` for sklearn >= 1.2 otherwise we import our version that is a copy-paste from sklearn.

In 1 year we will be dropping and always importing from sklearn.

Reply:
Got it 👍 Running now.